### PR TITLE
fix KeyError for multiple barcodes results possibly containing None

### PR DESCRIPTION
**Pull request by ArlindNocaj**
If you provide multiple images and some of them get None as a result from zxing, then the code failed with a Keyerror.
This got resolved by checking for these None values when generating the ordered list of responses.

### CODE REVIEW

File: zxing/__init__.py [hunk 1]
```diff
@@ -98,8 +98,8 @@ def decode(self, filenames, try_harder=False, possible_formats=None, pure_barcod
         else:
             # zxing (insanely) randomly reorders the output blocks, so we have to put them back in the
             # expected order, based on their URIs
-            d = {c.uri: c for c in codes}
-            return [d[f] for f in file_uris]
+            d = {c.uri: c for c in codes if c is not None} # there can be None in codes
+            return [d[f] if f in d else None for f in file_uris]
```
Comment (dlenski) [inline zxing/__init__.py:102]: This can be simplified to simply `return [d.get(f) for f in file_uris]`

**Review by dlenski — COMMENTED** [review]
Thanks.

It'd be good to add a test based on `test_decoding_multiple`, which includes not-parseable-as-barcode image(s) in order to verify this case is handled correctly.

https://github.com/dlenski/python-zxing/blob/master/test/test_all.py#L63-L93

**Comment by dlenski** [comment]
@ArlindNocaj, https://github.com/dlenski/python-zxing/commit/3c97cb4777e2744babd7d849dd3fadfb451908e1 should address the underlying issue you've identified.

Now, `BarCodeReader.decode` will return `BarCode(uri="whatever", raw=None, format=None)` (rather than `None`) when an image can't be parsed. Also includes tests to verify this.

**Comment by ArlindNocaj** [comment]
Thats great. 
In the meanwhile I created a library to process multiple barcodes on documents with jpype and zxing, without needing to parse text data: https://github.com/ArlindNocaj/document-barcodes

**Pull request closed without merging**